### PR TITLE
Add bootstrap_tenant procedure for tenant management

### PR DESCRIPTION
**Pull request by Junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Feature: https://github.com/datafuselabs/databend/issues/4529
Add admin$bootstrap_tenant procedure for tenant management.

## Changelog

- New Feature

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/4529

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/exception/src/exception_code.rs [hunk 1]
```diff
@@ -118,6 +118,7 @@ build_exceptions! {
     PermissionDenied(1063),
     UnmarshalError(1064),
     SemanticError(1065),
+    InvalidAuthInfo(1066),
```
Comment (flaneur2020) [inline common/exception/src/exception_code.rs:121]: what's the difference with AuthenticateFailure(1051)? q.q

File: common/meta/types/src/user_auth.rs [hunk 2]
```diff
@@ -104,7 +105,7 @@ fn double_sha1(v: &[u8]) -> [u8; 20] {
 }
 
 impl AuthInfo {
-    pub fn new(auth_type: AuthType, auth_string: &Option<String>) -> Result<AuthInfo, String> {
```
Comment (flaneur2020) [inline common/meta/types/src/user_auth.rs:107]: nice refactor!

File: query/src/users/role_mgr.rs [hunk 3]
```diff
@@ -48,12 +48,23 @@ impl UserApiProvider {
...
+        role_info: RoleInfo,
+        if_not_exists: bool,
+    ) -> Result<u64> {
         let client = self.get_role_api_client(tenant)?;
         let add_role = client.add_role(role_info);
         match add_role.await {
             Ok(res) => Ok(res),
-            Err(e) => Err(e.add_message_back("(while add role).")),
+            Err(e) => {
+                if if_not_exists && e.code() == ErrorCode::user_already_exists_code() {
```
Comment (flaneur2020) [inline query/src/users/role_mgr.rs:62]: maybe role_already_exists_code ?

Comment (Junnplus) [reply query/src/users/role_mgr.rs:62]: https://github.com/datafuselabs/databend/blob/908faae1f1f78844aa60fa3c4ac78f9b6e24b959/common/management/src/role/role_mgr.rs#L109

`RoleAlreadyExists` might be more reasonable.

File: common/exception/src/exception_code.rs (hunk 1 above)
Comment (Junnplus) [reply common/exception/src/exception_code.rs:121]: It is not used in the authentication process, exists for AuthInfo creation.

https://github.com/datafuselabs/databend/pull/4530/files#diff-50bc2112130914344462eb75dcd752843e2ca16edd6adcb3ba3a7b0589547f2eR45

**Review by ZhiHanZ — APPROVED** [review]
LGTM, we need to document the admin permissions which would be granted to bootstraped user

**Comment by BohuTANG** [comment]
> LGTM, we need to document the admin permissions which would be granted to bootstraped user

Document it in the databend.rs operation category ?

**Pull request merged**